source/azure_iot_provisioning_client.c

```diff
@@ -0,0 +1,963 @@
+            break;
+
+        case azureiotprovisioningWF_STATE_SUBSCRIBE:
+            xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState =
+                ulActionResult == eAzureIoTProvisioningSuccess ? azureiotprovisioningWF_STATE_SUBSCRIBING :
+                azureiotprovisioningWF_STATE_COMPLETE;
+            break;
+
+        case azureiotprovisioningWF_STATE_SUBSCRIBING:
+            xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState =
+                ulActionResult == eAzureIoTProvisioningSuccess ? azureiotprovisioningWF_STATE_REQUEST :
+                azureiotprovisioningWF_STATE_COMPLETE;
+            break;
+
+        case azureiotprovisioningWF_STATE_REQUEST:
+            xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState =
+                ulActionResult == eAzureIoTProvisioningSuccess ? azureiotprovisioningWF_STATE_REQUESTING :
+                azureiotprovisioningWF_STATE_COMPLETE;
+            break;
+
+        case azureiotprovisioningWF_STATE_REQUESTING:
+            xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState =
+                ulActionResult == eAzureIoTProvisioningSuccess ? azureiotprovisioningWF_STATE_RESPONSE :
+                azureiotprovisioningWF_STATE_COMPLETE;
+            break;
+
+        case azureiotprovisioningWF_STATE_RESPONSE:
+            xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState =
+                ulActionResult == eAzureIoTProvisioningPending ? azureiotprovisioningWF_STATE_WAITING :
+                azureiotprovisioningWF_STATE_COMPLETE;
+            break;
+
+        case azureiotprovisioningWF_STATE_WAITING:
+            xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState =
+                ulActionResult == eAzureIoTProvisioningSuccess ? azureiotprovisioningWF_STATE_REQUEST :
+                azureiotprovisioningWF_STATE_COMPLETE;
+            break;
+
+        case azureiotprovisioningWF_STATE_COMPLETE:
+            AZLogInfo( ( "Complete done\r\n" ) );
+            break;
+
+        default:
+            AZLogError( ( "Unknown state: %u", ulState ) );
+            configASSERT( false );
+            break;
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ *
+ * Implementation of connect action, this action is only allowed in azureiotprovisioningWF_STATE_CONNECT
+ *
+ * */
+static void prvAzureIoTProvisioningClientConnect( AzureIoTProvisioningClientHandle_t xAzureIoTProvisioningClientHandle )
+{
+    AzureIoTMQTTConnectInfo_t xConnectInfo = { 0 };
+    AzureIoTProvisioningClientResult_t xReturn;
+    AzureIoTMQTTResult_t xResult;
+    uint8_t ucSessionPresent;
+    uint32_t ulBytesCopied = 0;
+    size_t xMQTTUsernameLength;
+    az_result xCoreResult;
+
+    if( xAzureIoTProvisioningClientHandle->_internal.ulWorkflowState != azureiotprovisioningWF_STATE_CONNECT )
+    {
+        AZLogWarn( ( "Connect action called in wrong state %d\r\n",
```

Comment:
Only reason is the COM port client (PUTTY). Because they generally have some configuration to \n -> newline.

---
In reply to: [602576961](https://github.com/Azure/azure-iot-middleware-freertos/pull/49#discussion_r602576961) [](ancestors = 602576961)

Reply:
I will remove the \r\n from logs and let the log implementation to add new line.

---
In reply to: [603467682](https://github.com/Azure/azure-iot-middleware-freertos/pull/49#discussion_r603467682) [](ancestors = 603467682,602576961)

Reply:
"I will remove the \r\n from logs and let the log implementation to add new line."
Yeah, I think that's fair and if someone had a platform it mattered they could always tweak there.  It's a very minor perf hit on 2 I/O calls presumably but it's not that bad and logging doesn't promise ultra high perf anyway.

I took a quick look through FreeRTOS source and they're all over the place about consistency - I even saw a "\n\r" which in +20 years of doing I've never seen before:).